File: objectModel/CSharp/Microsoft.CommonDataModel.ObjectModel/Cdm/StringSpewCatcher.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System.Runtime.CompilerServices;
using System.Text;

#if INTERNAL_VSTS
[assembly: InternalsVisibleTo("Microsoft.CommonDataModel.ObjectModel.Tests" + Microsoft.CommonDataModel.AssemblyRef.TestPublicKey)]
#else
[assembly: InternalsVisibleTo("Microsoft.CommonDataModel.ObjectModel.Tests")]
#endif
namespace Microsoft.CommonDataModel.ObjectModel.Cdm
{
    internal class StringSpewCatcher
    {
        private StringBuilder content = new StringBuilder();
        private StringBuilder segment = new StringBuilder();

        public void Clear()
        {
            this.content.Clear();
            this.segment.Clear();
        }

        public void SpewLine(string spew)
        {
            this.segment.Append(spew + "\n");
            if (this.segment.Length > 1000)
            {
                this.content.Append(this.segment);
                this.segment.Clear();
            }
        }

        public string GetContent()
        {
            return this.content.Append(this.segment).ToString();
        }
    }
}

File: objectModel/Java/objectmodel/src/main/java/com/microsoft/commondatamodel/objectmodel/cdm/StringSpewCatcher.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

package com.microsoft.commondatamodel.objectmodel.cdm;

public class StringSpewCatcher {
  private StringBuilder content = new StringBuilder();
  private StringBuilder segment = new StringBuilder();

  public void clear() {
    this.content = new StringBuilder();
    this.segment = new StringBuilder();
  }

  public void spewLine(final String spew) {
    this.segment.append(spew).append("\n");
    if (this.segment.length() > 1000) {
      this.content.append(this.segment);
      this.segment = new StringBuilder();
    }
  }

  public String getContent() {
    return this.content.append(this.segment).toString();
  }
}
